File: go/find-the-highest-altitude.go
func largestAltitude(gain []int) int {
    currentAltitude := 0
    maximalAltitude := 0
    for i := 0; i < len(gain); i++ {
        currentAltitude += gain[i]
        if maximalAltitude < currentAltitude {
            maximalAltitude = currentAltitude
        }
    }
    return maximalAltitude
}

File: python/find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        current_altitude = 0
        maximal_altitude = 0
        for g in gain:
            current_altitude += g
            maximal_altitude = max(maximal_altitude, current_altitude)
        return maximal_altitude
